test/spec/ol/interaction/drawboxinteraction.test.js

```diff
@@ -0,0 +1,338 @@
+      simulateEvent('pointerup', 10, 20);
+
+      // finish with second point
+      simulateEvent('pointermove', 20, 30);
+      simulateEvent('pointerdown', 20, 30);
+      simulateEvent('pointerup', 20, 30);
+
+      expect(ds).to.be.called(1);
+      expect(de).to.be.called(1);
+    });
+
+  });
+
+  describe('drawing squares', function() {
+    var draw;
+
+    beforeEach(function() {
+      draw = new ol.interaction.DrawBox({
+        source: source,
+        condition: ol.events.condition.always
+      });
+      map.addInteraction(draw);
+    });
+
+    it('draws a square with clicks', function() {
+      // first point
+      simulateEvent('pointermove', 10, 20);
+      simulateEvent('pointerdown', 10, 20);
+      simulateEvent('pointerup', 10, 20);
+
+      // finish with second point
+      simulateEvent('pointermove', 20, 32);
+      simulateEvent('pointerdown', 20, 32);
+      simulateEvent('pointerup', 20, 32);
+
+      var features = source.getFeatures();
+      expect(features).to.have.length(1);
+      var geometry = features[0].getGeometry();
+      expect(geometry).to.be.a(ol.geom.Polygon);
+
+      expect(geometry.getCoordinates()).to.eql([
+        [[10, -32], [10, -20], [22, -20], [22, -32]]
+      ]);
+    });
+
+    it('triggers draw events', function() {
+      var ds = sinon.spy();
+      var de = sinon.spy();
+      goog.events.listen(draw, ol.DrawEventType.DRAWSTART, ds);
+      goog.events.listen(draw, ol.DrawEventType.DRAWEND, de);
+
+      // first point
+      simulateEvent('pointermove', 10, 20);
+      simulateEvent('pointerdown', 10, 20);
+      simulateEvent('pointerup', 10, 20);
+
+      // finish with second point
+      simulateEvent('pointermove', 20, 30);
+      simulateEvent('pointerdown', 20, 30);
+      simulateEvent('pointerup', 20, 30);
+
+      expect(ds).to.be.called(1);
+      expect(de).to.be.called(1);
+    });
+
+  });
+
+  describe('#setActive()', function() {
+    var interaction;
+
+    beforeEach(function() {
+      interaction = new ol.interaction.DrawBox({
+        source: source
+      });
+
+      expect(interaction.getActive()).to.be(true);
+
+      map.addInteraction(interaction);
+
+      // first point
+      simulateEvent('pointermove', 10, 20);
+      simulateEvent('pointerdown', 10, 20);
+      simulateEvent('pointerup', 10, 20);
+
+      expect(interaction.sketchFeature_).not.to.be(null);
+    });
+
+    afterEach(function() {
+      map.removeInteraction(interaction);
+    });
+
+    describe('#setActive(false)', function() {
+      it('unsets the map from the feature overlay', function() {
+        interaction.setActive(false);
+        expect(interaction.overlay_.map_).to.be(null);
+      });
+      it('aborts the drawing', function() {
+        interaction.setActive(false);
+        expect(interaction.sketchFeature_).to.be(null);
+      });
+      it('fires change:active', function() {
+        listenerSpy = sinon.spy(function() {
```

Comment:
This needs a `var` to avoid leaking a global (I see that `drawinteraction.test.js` has the same problem and will fix separately).  I'll also make some time before too long to update our linter to catch this.